src/peeker/src/main.rs

```diff
@@ -110,7 +131,13 @@ fn spawn_query_thread(mz_url: String, query_group: QueryGroup) -> Vec<JoinHandle
             let mut counter = 0u64;
             let mut err_count = 0u64;
             let mut last_log = Instant::now();
+            let end_at = Utc::now() + run_dur;
+            info!(
+                "Will terminate thread {}.{} at {:?}",
+                group_id, thread_id, end_at
+            );
             loop {
+                let mut exit_loop = false;
```

Comment:
Tioli: Not sure if it's worth using a feature as obscure as this, but you could break out off this loop with a label:  https://doc.rust-lang.org/stable/rust-by-example/flow_control/loop/nested.html

Reply:
Ah--I really like that! ty